Adding support for AsyncAws

Nyholm opened the pull request:
This will fix #638

Nyholm commented on tests/Gaufrette/Functional/Adapter/AsyncAwsS3Test.php:
```diff
@@ -0,0 +1,155 @@
+
+    protected function tearDown()
+    {
+        // TODO delete bucket
```
True, I added that in the S3 client after I made this PR. 

I should update this.

Nyholm commented:
Friendly ping for @nicolasmure and @nm2107 

Do you need help maintaining Gaufrette and GaufretteBundle?

alexpozzi commented:
Hello @Nyholm,
@nicolasmure and @nm2107 (which are the same person :smile:) are not maintaining this library anymore so help is very welcome.
I can grant you access to this repository and the associated SF bundle if it's okay with you.

Nyholm commented:
Sure. Thank you.

alexpozzi commented:
Done, don't hesitate to ping me for help or assign me code reviews, I'll do my best to be reactive :wink: 

Thanks for your help!

Nyholm commented:
Thank you. 

I've made sure master is green and supports PHP7.4. I've also added some tests on github actions. I removed some deprecated code. There is no point in keeping that around for long. 

I'll merge this and test on PHP8. After that I'll release 0.10.

stof commented on src/Gaufrette/Adapter/AsyncAwsS3.php:
```diff
@@ -0,0 +1,339 @@
+     */
+    public function __construct(SimpleS3Client $service, $bucket, array $options = [], $detectContentType = false)
+    {
+        if (!class_exists(SimpleS3Client::class)) {
```
this check is useless. If the class does not exist, it is impossible to pass an argument to `$service` due to the typehint. So this is dead code.

Nyholm replied:
That is true. 

Thank you
